mtags/src/main/scala-3/scala/meta/internal/pc/SignatureHelpProvider.scala

```diff
@@ -0,0 +1,163 @@
+package scala.meta.internal.pc
+
+import scala.collection.JavaConverters.*
+
+import scala.meta.internal.mtags.MtagsEnrichments.*
+import scala.meta.pc.OffsetParams
+
+import dotty.tools.dotc.ast.Trees.UnApply
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.tpd.Tree
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Denotations.SingleDenotation
+import dotty.tools.dotc.core.Symbols
+import dotty.tools.dotc.core.Types.ErrorType
+import dotty.tools.dotc.core.Types.MethodType
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.reporting.AmbiguousOverload
+import dotty.tools.dotc.reporting.NoMatchingOverload
+import dotty.tools.dotc.util.Signatures
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.util.Spans.Span
+import org.eclipse.lsp4j.ParameterInformation
+import org.eclipse.lsp4j.SignatureHelp
+import org.eclipse.lsp4j.SignatureInformation
+
+object SignatureHelpProvider:
+
+  def contribute(driver: InteractiveDriver, params: OffsetParams) =
+    val uri = params.uri
+    val sourceFile = CompilerInterfaces.toSource(params.uri, params.text)
+    driver.run(uri, sourceFile)
+
+    given ctx: Context = driver.currentCtx
+    // given locatedCtx: Context = driver.localContext(params)
+    val pos = driver.sourcePosition(params)
+    val trees = driver.openedTrees(uri)
+    val path =
+      Interactive
+        .pathTo(trees, pos)(using ctx)
+        .dropWhile {
+          case _: tpd.Apply => false
+          case unapply: tpd.UnApply =>
+            // Don't show tuple unapply method
```

Comment:
What happens if we show the tuple unapply method?

Reply:
+1

Reply:
It's just not really useful, since when writing tuples we are not really interested in signature help.